File: Day 5/ZeroFuel.java
// WPU Coding Challenge 2024
// 5/366
// https://www.codewars.com/kata/5861d28f124b35723e00005e

public class ZeroFuel {
  public static boolean zeroFuel(double distanceToPump, double mpg, double fuelLeft) {
    return distanceToPump <= mpg * fuelLeft;
  }

  public static void main(String[] args) {
    System.out.println(zeroFuel(50, 25, 2));
  }
}

File: Day 5/zeroFuel.kt
// WPU Coding Challenge 2024
// 5/366
// https://www.codewars.com/kata/5861d28f124b35723e00005e

fun main() {
    println(zeroFuel(50, 25, 2))
    println(zeroFuel(100, 50, 1))
}

// Solution logic: the distance of nearest pump divided by car miles per gallon should be minimum equal to the gallon fuel left in the car
fun zeroFuel(distanceToPump: Int, mpg: Int, fuelLeft: Int): Boolean = distanceToPump / mpg <= fuelLeft

// Solution logic: the distance of nearest pump should be less than or equal to car miles per gallon times the gallon fuel left in the car
fun zeroFuel2(distanceToPump: Int, mpg: Int, fuelLeft: Int): Boolean = distanceToPump <= mpg * fuelLeft